src/Locator/GitRepositoryDirLocator.php

```diff
@@ -36,7 +39,22 @@ public function locate(string $gitDir): string
         $gitRepositoryDir = $matches[1];
 
         if ($this->filesystem->isAbsolutePath($gitRepositoryDir)) {
```

Comment:
I was a bit confused whilst trying to figure out how this works exactly. (took around 30 minutes to click)

Would you mind moving this logic into a private function and add documentation on how this logic works in that function's docblock?

Reply:
Extracted and added comments